Upgrade fails when server omits reason phrase

Invoking nghttp with the `--upgrade` flag does not work if the server omits the optional reason-phrase from the status-line. (This is the [default behavior](https://bz.apache.org/bugzilla/show_bug.cgi?id=60362) of Tomcat starting with 8.5. Tomcat 9.0 cannot be configured to send the reason-phrase at all.) For example:

```bash
 ~ $ nghttp --no-dep -v 'http://[::1]:8080/' -u
[  0.000] Connected
[  0.000] HTTP Upgrade request
GET / HTTP/1.1
host: [::1]:8080
connection: Upgrade, HTTP2-Settings
upgrade: h2c
http2-settings: AAMAAABkAAQAAP__
accept: */*
user-agent: nghttp2/1.23.1


[  0.001] HTTP Upgrade response
HTTP/1.1 101
Connection: Upgrade
Upgrade: h2c
Date: Sun, 25 Jun 2017 19:06:41 GMT


[ERROR] HTTP Upgrade failed
Some requests were not processed. total=1, processed=0
```

The omission of the `Switching Protocols` text breaks nghttp2's request parsing in such a way that it thinks the Upgrade request failed. The root cause is in `third-party/http-parser/http_parser.c`: skipping the reason-phrase causes the `s_res_status` parser state to be skipped, which in turn means that the `on_status` callback (namely `htp_statuscb`) never fires, which means that `upgrade_response_status_code` is never set; any status code other than 101 is (reasonably) interpreted as failure:

```c++
  if (upgrade_response_status_code != 101) {
    std::cerr << "[ERROR] HTTP Upgrade failed; status was " << upgrade_response_status_code << std::endl;

    return -1;
  }
```

--- a/src/nghttp.cc
+++ b/src/nghttp.cc
@@ -405,16 +405,9 @@ int htp_msg_begincb(http_parser *htp) {
 } // namespace
 
 namespace {
-int htp_statuscb(http_parser *htp, const char *at, size_t length) {
-  auto client = static_cast<HttpClient *>(htp->data);
-  client->upgrade_response_status_code = htp->status_code;
-  return 0;
-}
-} // namespace
-
-namespace {
 int htp_msg_completecb(http_parser *htp) {
   auto client = static_cast<HttpClient *>(htp->data);
+  client->upgrade_response_status_code = htp->status_code;
   client->upgrade_response_complete = true;
   return 0;
 }
@@ -424,7 +417,7 @@ namespace {
 constexpr http_parser_settings htp_hooks = {
     htp_msg_begincb,   // http_cb      on_message_begin;
     nullptr,           // http_data_cb on_url;
-    htp_statuscb,      // http_data_cb on_status;
+    nullptr,           // http_data_cb on_status;
     nullptr,           // http_data_cb on_header_field;
     nullptr,           // http_data_cb on_header_value;
     nullptr,           // http_cb      on_headers_complete;

